client/src/main/java/io/pravega/client/security/auth/DelegationTokenProviderImpl.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.security.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.stream.impl.Controller;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DelegationTokenProviderImpl implements DelegationTokenProvider {
+
+    /**
+     * The delegate.
+     */
+    @VisibleForTesting
+    @Getter(AccessLevel.PACKAGE)
+    private final DelegationTokenHandlingStrategy strategy;
+
+    /**
+     * Constructs a new object with the specified specified delegation token.
+     *
+     * <p>
+     * Intended for use mainly for testing purposes.
+     */
+    @VisibleForTesting
+    DelegationTokenProviderImpl(@NonNull String token) {
+        if (token.trim().equals("")) {
+            strategy = new EmptyTokenHandlingStrategy();
+            log.debug("Set DelegationTokenHandlingStrategy as {}", EmptyTokenHandlingStrategy.class);
```

Comment:
Please do not log every time you create an object. This will pollute the log files like crazy. Move your logging to the upstream code that instantiates this.

Reply:
I was logging these at a debug level and it's going to be in the client logs. In any case, I can't help but agree that these were logging way too much. I've now removed these debug statements.